### PR TITLE
feat(unlock-app) refactor and add validation to multiple recipients

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
This PR adds new validation for multiple recipient hook, in detail:
- Check for duplicates in list
- Check if address already owns a key

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/hooks/useMultipleRecipient.ts [hunk 1]
```diff
@@ -164,12 +178,23 @@ export const useMultipleRecipient = (
...
         )
+        ToastHelper.success('Recipient correctly added in list.')
       }
-      if (!valid) {
+
+      if (isAddressWithKey) {
+        ToastHelper.error(
+          'This address already owns a valid key. You cannot grant them a new one.'
+        )
+      } else if (isAddressInList) {
```
Comment (searchableguy) [inline unlock-app/src/hooks/useMultipleRecipient.ts:188]: But we allow multiple keys now with v10 lock, right?

File: unlock-app/src/hooks/useMultipleRecipient.ts [hunk 2]
```diff
@@ -89,15 +91,43 @@ export const useMultipleRecipient = (
...
+    const canAddMultiple = (lock?.publicLockVersion ?? 1) >= 10
+
+    // todo: need also to check how many keys the address owns to improve this logic
+    const limitNotReached = addressItemsCount < (lock?.maxKeysPerAddress ?? 1)
+
+    const valid =
+      address?.length > 0 &&
+      ((canAddMultiple && limitNotReached) ||
+        (!isAddressInList && !isAddressWithKey))
+
```
Comment (searchableguy) [inline unlock-app/src/hooks/useMultipleRecipient.ts:117]: Can deconstruct it into multiple lines for clarity? It's confusing.

File: unlock-app/src/hooks/useMultipleRecipient.ts [hunk 3]
```diff
@@ -27,19 +28,20 @@ interface RecipientPayload {
 }
 
 export const useMultipleRecipient = (
-  paywallConfig?: PaywallConfig,
-  lockAddress?: string
+  lock: Lock,
+  network?: number,
+  maxRecipients = 1
 ) => {
+  console.log('here lock', lock)
```
Comment (julien51) [inline unlock-app/src/hooks/useMultipleRecipient.ts:35]: 🧹

File: unlock-app/src/hooks/useMultipleRecipient.ts [hunk 4]
```diff
@@ -89,15 +91,43 @@ export const useMultipleRecipient = (
 
   const getAddressAndValidation = async (recipient: string) => {
     const address = await getAddressForName(recipient)
+    const existingExpiration = await web3Service.getKeyExpirationByLockForOwner(
```
Comment (julien51) [inline unlock-app/src/hooks/useMultipleRecipient.ts:94]: Let's use `getHasValidKey` because it's compatible with all versions :)

Comment (julien51) [reply unlock-app/src/hooks/useMultipleRecipient.ts:94]: This way you don't have to deal with the expiration logic , which simplifies things!

**Review by julien51 — COMMENTED** [review]
A small change but this is the right direction!

**Review by searchableguy — APPROVED** [review]

**Pull request merged**